Initial draft of Variational Inference (#47)
MrLongbottom/P9-Rapport: sections/appendix/variational_inference.tex

\subsection{\gls{vi}}

> When using topic models and other Bayesian models, it is computationally infeasable to compute the posterior, which means approximation is needed. 

Reviewer: ```suggestion
When using topic models and other Bayesian models, it is computationally infeasible to compute the posterior, which means approximation is necessary. 
```

We do the approximation by fitting variational parameters $\nu$, to mimize the distance between initial distribution $v$ and $\nu^*$ where $\nu^*$ is the distribution that is closest to $p(z|\textbf{x})$.
In \autoref{fig:vi}, we can see a visualization of this where we want to optimize the variational parameters to minimize the distance be the true posterior distribution.
In \autoref{fig:vi}, $q(z; \nu)$ is a family of variational distributions over the latent variables which is different instances of the hidden variables $z$.
Our goal is to fit the variational paramaters $\nu$ to estimate the hidden variables that are closest to $p(z|\textbf{x})$

\begin{figure}
	\centering
	\includegraphics[width=0.5\textwidth]{figures/vi_illustration.png}
	\caption[Caption for LOF]{Illustration of \acrlong{vi}\footnotemark}
	\label{fig:vi}
\end{figure}
\footnotetext{http://www.cs.columbia.edu/~blei/talks/Blei\_VI\_tutorial.pdf}

\subsubsection*{LDA learning}
We use an implementation of the LDA which is based on \cite{blei2010online}, where \citeauthor{blei2010online} present a way of training the LDA using \gls{vi}.

> They do this by using Stocastic \gls{vi}, which also outperforms batch learning.

Reviewer 2: ```suggestion
They do this by using stocastic \gls{vi}, which also outperforms batch learning.
```

> In learning the \gls{lda}, we first sample the whole corpus, where we learn the structure of this sample and use this to change parameters in the global model.

Reviewer 2: global model er et nyt begreb

> In learning the \gls{lda}, we first sample the whole corpus, where we learn the structure of this sample and use this to change parameters in the global model.

Author: ```suggestion
In learning the \gls{lda}, we first sample the whole corpus, where we learn the structure of this sample and use this to change parameters in the global parameters.
```

To do this, we need some form of verification of whether we are getting closer to the the true posterior distribution $p(z|\textbf{x})$.

> In \gls{lda}, we use perplexity as a score to check whether, we are moving towards $p(z|\textbf{x})$

Reviewer 2: ```suggestion
In \gls{lda}, we use perplexity as a score to check whether we are moving towards $p(z|\textbf{x})$
```

We do the approximation by fitting variational parameters $\nu$, to minimize the distance between the initial distribution $v$ and $\nu^*$, where $\nu^*$ is the distribution that is closest to $p(z|\textbf{x})$.
In \autoref{fig:vi}, we can see a visualization of this where we want to optimize the variational parameters to minimize the distance to the true posterior distribution.
In \autoref{fig:vi}, $q(z; \nu)$ is a family of variational distributions over the latent variables which are different instances of the hidden variables $z$.
Our goal is to fit the variational parameters $\nu$ to estimate the hidden variables that are closest to $p(z|\textbf{x})$

\begin{figure}
	\centering
	\includegraphics[width=0.5\textwidth]{figures/vi_illustration.png}
	\caption[Caption for LOF]{Illustration of \acrlong{vi}\footnotemark.}
	\label{fig:vi}
\end{figure}
\footnotetext{http://www.cs.columbia.edu/~blei/talks/Blei\_VI\_tutorial.pdf}

\subsubsection*{LDA learning}
We use an implementation of the \gls{lda} which is based on \cite{blei2010online}, where \citeauthor{blei2010online} present a way of training the \gls{lda} using \gls{vi}.

> They do this by using stocastic \gls{vi}, which also outperforms batch learning.

Reviewer 2: ```suggestion
They do this by using stochastic \gls{vi}, which also outperforms batch learning.
```
